PR: Tabify Outline next to Projects in default layout and other layout improvements and fixes

ccordoba12 opened the pull request:
## Description of Changes

- While translating our docs to Spanish, @LucyJimenez noted that the Outline is tabified next to Projects, but that's not the case for the default layout. So, this PR changes that, which I think is a good idea.
- For consistency, show Outline to the left when the editor is maximized and in editor windows.
- Fix setting any layout with the correct proportions declared in its definition. That was not working as expected, which made the Outline/Projects column in the default layout appear too thin.
- Fix bug that prevented to restore the previous layout at startup for some user defined layouts (see #21596).
- Fix bug that broke positioning Spyder in the right screen at startup. That was a regression introduced in #21572.
- Allow to run Spyder with PyQt6 or PySide6 from `bootstrap.py`. I did this to test that the fix to position Spyder in the right screen worked for both Qt5 and 6.
- Give editor windows a good size independent from the `EditorMainWidget` one. That's in case the latter is too small.

### Issue(s) Resolved

<!--- List the issue(s) below, in the form "Fixes #1234"; one per line --->

Fixes #22178
Fixes #21596

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

<!--- TYPE YOUR USER/NAME AFTER THE FOLLOWING: --->
I certify the above statement is true and correct: @ccordoba12 

<!--- Thanks for your help making Spyder better for everyone! --->

ccordoba12 commented:
@dalthviz, this is ready for review.

ccordoba12 commented on spyder/app/mainwindow.py:
```diff
@@ -845,15 +845,6 @@ def pre_visible_setup(self):
         if self.layouts is not None:
             self.layouts.register_custom_layouts()
 
-        # Needed to ensure dockwidgets/panes layout size distribution
-        # when a layout state is already present.
-        # See spyder-ide/spyder#17945
-        if (
-            self.layouts is not None
-            and self.get_conf('window/state', default=None)
-        ):
-            self.layouts.before_mainwindow_visible()
```
This bit of code was what actually generated issue #21596 for the layout distribution shown in the OP. That's why I removed it.

However, I checked that issue #17945 was not triggered again by this removal. But please check if that's the case for you too.

dalthviz replied:
I was able to reproduce #17945 by launching Spyder from the work on this PR :/ :

![layout_distribution](https://github.com/spyder-ide/spyder/assets/16781833/d0f618af-8241-45f5-be37-5a20e2364213)

|Initial distribution with maximum height given to the IPython Console|Final distribution after closing and launching Spyder multiple times|
|-|-|
|![image](https://github.com/spyder-ide/spyder/assets/16781833/1ca636f5-829f-403c-8001-cd752b625f69)|![image](https://github.com/spyder-ide/spyder/assets/16781833/dde2e871-dcf1-4aae-b406-e24f22783edb)|

ccordoba12 replied:
Ok, it's very weird but I was easily able to reproduce the problem today (I thought I have tested it and it wasn't happening anymore).

However, I amended one of my last commits and now it should be working as expected (both for issue #17945 and #21596). So, please test again.

dalthviz replied:
Checked again and seems like indeed the latest changes help with both issues :+1:

ccordoba12 commented on spyder/app/mainwindow.py:
```diff
@@ -1140,16 +1131,13 @@ def closing(self, cancelable=False, close_immediately=False):
         # Fixes spyder-ide/spyder#12139
+        self.plugin_registry.dock_all_undocked_plugins(save_undocked=True)
+
+        # Save layout before closing all plugins. This ensures its restored
+        # correctly in the next session when there are many IPython consoles
+        # open in the current one.
```
I think this change makes saving the layout more robust.

dalthviz commented:
Checking this I noticed the following (although not completely sure if the changes here are the cause and some are just ideas/questions that occured to me):

* Outline on an Editor in an independent window doesn't hide:
![hide_outline](https://github.com/user-attachments/assets/b0850dd7-4d9a-4a71-904d-c0f0ebd62bb8)

* Checking things with PyQt6 (although seems like with PyQt5 too - maybe something messed up in my setup 🤔):
  * Seems like if you  snap Spyder to use half of the screen, after relaunching the main window ends up with strange proportions:
  ![proportions_pyqt6](https://github.com/user-attachments/assets/b5932cb6-4cf7-451d-9d66-7084e45e9024)
  * Seems like there is a message related with an image:
    `<use> element image0 in wrong context!`
   ![image](https://github.com/user-attachments/assets/5df46c0f-f379-4c21-9e8d-85ebcb95ad9b)


* Related with the `--gui` option, should it be worthy to have some documentation for that? Maybe over https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md#running-spyder ? Or even documentation somewhere where the difference between `python bootstrap.py --<something>` vs `python bootstrap.py -- --<something>` is explained?

* This is marked as solving https://github.com/spyder-ide/spyder/issues/22178 maybe the changes to do that are making the Linux CI slow jobs to fail?

ccordoba12 commented:
Thanks for the additional review @dalthviz! About your comments:

> Outline on an Editor in an independent window doesn't hide

Great catch! I didn't test that when moving the Outline to the left but it should be fixed now.

> Seems like if you snap Spyder to use half of the screen, after relaunching the main window ends up with strange proportions

Great catch too! I was easily able to reproduce this by simply closing the main window when is not maximized.

The solution was to restore the code to apply the layout twice: before and after the main window is visible (in my previous solution I though it was necessary to apply it only after). So, the fix you provided for issue #17945 was almost right, but in order to address issue #21596, the second time the layout is applied needs to happen after the `View > Panes` menu is created.

> Seems like there is a message related with an image

Fixed in my last commit.

> Related with the --gui option, should it be worthy to have some documentation for that? Maybe over https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md#running-spyder ? Or even documentation somewhere where the difference between python bootstrap.py --<something> vs python bootstrap.py -- --<something> is explained?

I addressed that in my last commit too.

> This is marked as solving https://github.com/spyder-ide/spyder/issues/22178 maybe the changes to do that are making the Linux CI slow jobs to fail?

The errors on CI were happening because I removed the code to apply the layout before the main window is visible. And I decided to fix #22178 here because while looking at the CI logs I saw that many tests were flagged as having errors due to that issue.

dalthviz commented on spyder/plugins/layout/api.py:
```diff
@@ -496,6 +504,13 @@ def set_main_window_layout(self, main_window, dockable_plugins):
+            for plugin in base_plugins_to_hide:
+                plugin.toggle_view(False)
+
+        QTimer.singleShot(50, hide_not_visible_base_plugins)
```
Why is this timer needed?

dalthviz commented on spyder/plugins/outlineexplorer/plugin.py:
```diff
@@ -149,8 +149,20 @@ def dock_with_maximized_editor(self):
+            editor = self.get_plugin(Plugins.Editor)
+            self.main.resizeDocks(
+                [editor.dockwidget, self.dockwidget],
+                [self.main.width(), min(self.main.width() // 7, max_width)],
```
Any particular reason to choose `7` here?

dalthviz commented on spyder/plugins/layout/plugin.py:
```diff
@@ -532,7 +521,7 @@ def load_window_settings(self, prefix, default=False, section='main'):
         width = pos[0]
         height = pos[1]
-        screen_shape = self.main.screen().geometry()
+        screen_shape = self.main.screen().virtualGeometry()
```
Maybe describing the advantage or need for using `virtualGeometry` instead of `geometry` here could be worthy

dalthviz reviewed (approved):
Thanks @ccordoba12 ! Gave this last check and seems like things are working as expected. Left a couple of comments but I will leave this PR approved so feel free to merge it when you think is ready :+1:

Also, on an, I think, unrelated note, while checking this I started to notice that the options menus of the plugins panes need to be requested two times for them to render correctly :thinking: I don't think that is related with the work on this PR but seemed a behavior worthy of mention:

![options_menu_render](https://github.com/user-attachments/assets/3b61105f-e52e-4cf8-a415-fb47f06cf5b0)

ccordoba12 commented:
> Left a couple of comments but I will leave this PR approved so feel free to merge it when you think is ready 👍

Thanks! I addressed them in my last commit.

>  Also, on an, I think, unrelated note, while checking this I started to notice that the options menus of the plugins panes need to be requested two times for them to render correctly 🤔 I don't think that is related with the work on this PR but seemed a behavior worthy of mention

Are you observing that with PyQt 5 or 6? I can easily reproduce it with Qt 6 but not with 5. If that's the case for you too, we should leave it for later (but please open an issue about it).

dalthviz commented:
> Are you observing that with PyQt 5 or 6? I can easily reproduce it with Qt 6 but not with 5. If that's the case for you too, we should leave it for later (but please open an issue about it).

I'm able to see the behavior with PyQt5 using master. My setup info:

* Spyder version: 6.0.0b3.dev0 e1fc1a987 (conda)
* Python version: 3.9.19 64-bit
* Qt version: 5.15.8
* PyQt5 version: 5.15.9
* Operating System: Windows-10-10.0.19045-SP0
* Conda list (my dev env):

<details>

```
# packages in environment at C:\Users\dalth\anaconda3\envs\spyder-dev:
#
# Name                    Version                   Build  Channel
aiohttp                   3.9.5            py39ha55989b_0    conda-forge
aiosignal                 1.3.1              pyhd8ed1ab_0    conda-forge
alabaster                 0.7.16             pyhd8ed1ab_0    conda-forge
arrow                     1.3.0              pyhd8ed1ab_0    conda-forge
astroid                   3.2.2            py39hcbf5309_0    conda-forge
asttokens                 2.4.1              pyhd8ed1ab_0    conda-forge
async-timeout             4.0.3              pyhd8ed1ab_0    conda-forge
asyncssh                  2.14.1             pyhd8ed1ab_0    conda-forge
atomicwrites              1.4.1              pyhd8ed1ab_0    conda-forge
attrs                     23.2.0             pyh71513ae_0    conda-forge
autopep8                  2.0.4              pyhd8ed1ab_0    conda-forge
babel                     2.14.0             pyhd8ed1ab_0    conda-forge
backports                 1.0                pyhd8ed1ab_3    conda-forge
backports.tarfile         1.0.0              pyhd8ed1ab_1    conda-forge
bcrypt                    4.1.3            py39h92a245a_0    conda-forge
beautifulsoup4            4.12.3             pyha770c72_0    conda-forge
binaryornot               0.4.4                      py_1    conda-forge
black                     24.4.2           py39hcbf5309_0    conda-forge
bleach                    6.1.0              pyhd8ed1ab_0    conda-forge
brotli                    1.1.0                hcfcfb64_1    conda-forge
brotli-bin                1.1.0                hcfcfb64_1    conda-forge
brotli-python             1.1.0            py39h99910a6_1    conda-forge
bzip2                     1.0.8                hcfcfb64_5    conda-forge
ca-certificates           2024.6.2             h56e8100_0    conda-forge
certifi                   2024.6.2           pyhd8ed1ab_0    conda-forge
cffi                      1.16.0           py39ha55989b_0    conda-forge
chardet                   5.2.0            py39hcbf5309_1    conda-forge
charset-normalizer        3.3.2              pyhd8ed1ab_0    conda-forge
click                     8.1.7           win_pyh7428d3b_0    conda-forge
cloudpickle               3.0.0              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
comm                      0.2.2              pyhd8ed1ab_0    conda-forge
contourpy                 1.2.1            py39h1f6ef14_0    conda-forge
cookiecutter              2.6.0              pyhca7485f_0    conda-forge
coverage                  6.5.0            py39ha55989b_1    conda-forge
cryptography              42.0.8           py39hae46aff_0    conda-forge
cycler                    0.12.1             pyhd8ed1ab_0    conda-forge
cython                    0.29.32          py39h99910a6_1    conda-forge
debugpy                   1.8.1            py39h99910a6_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
deprecated                1.2.14             pyh1a96a4e_0    conda-forge
diff-match-patch          20230430           pyhd8ed1ab_0    conda-forge
dill                      0.3.8              pyhd8ed1ab_0    conda-forge
docstring-to-markdown     0.15               pyhd8ed1ab_0    conda-forge
docutils                  0.21.2             pyhd8ed1ab_0    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
exceptiongroup            1.2.0              pyhd8ed1ab_2    conda-forge
executing                 2.0.1              pyhd8ed1ab_0    conda-forge
flake8                    7.0.0              pyhd8ed1ab_0    conda-forge
flaky                     3.7.0              pyh9f0ad1d_0    conda-forge
fonttools                 4.53.0           py39ha55e580_0    conda-forge
freetype                  2.12.1               hdaf720e_2    conda-forge
frozenlist                1.4.1            py39ha55989b_0    conda-forge
fzf                       0.53.0               hd02998f_0    conda-forge
gettext                   0.22.5               h5728263_2    conda-forge
gettext-tools             0.22.5               h7d00a51_2    conda-forge
glib                      2.80.2               h7025463_1    conda-forge
glib-tools                2.80.2               h4394cf3_1    conda-forge
gst-plugins-base          1.22.9               h001b923_1    conda-forge
gstreamer                 1.22.9               hb4038d2_1    conda-forge
icu                       73.2                 h63175ca_0    conda-forge
idna                      3.7                pyhd8ed1ab_0    conda-forge
imagesize                 1.4.1              pyhd8ed1ab_0    conda-forge
importlib-metadata        7.1.0              pyha770c72_0    conda-forge
importlib-resources       6.4.0              pyhd8ed1ab_0    conda-forge
importlib_metadata        7.1.0                hd8ed1ab_0    conda-forge
importlib_resources       6.4.0              pyhd8ed1ab_0    conda-forge
inflection                0.5.1              pyh9f0ad1d_0    conda-forge
iniconfig                 1.1.1              pyh9f0ad1d_0    conda-forge
intel-openmp              2024.1.0           h57928b3_966    conda-forge
intervaltree              3.1.0              pyhd8ed1ab_1    conda-forge
ipykernel                 6.29.4             pyh4bbf305_0    conda-forge
ipython                   8.18.1             pyh7428d3b_3    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
isort                     5.13.2             pyhd8ed1ab_0    conda-forge
jaraco.classes            3.4.0              pyhd8ed1ab_1    conda-forge
jaraco.context            5.3.0              pyhd8ed1ab_1    conda-forge
jaraco.functools          4.0.0              pyhd8ed1ab_0    conda-forge
jedi                      0.19.1             pyhd8ed1ab_0    conda-forge
jellyfish                 1.0.4            py39h92a245a_0    conda-forge
jinja2                    3.1.4              pyhd8ed1ab_0    conda-forge
jsonschema                4.22.0             pyhd8ed1ab_0    conda-forge
jsonschema-specifications 2023.12.1          pyhd8ed1ab_0    conda-forge
jupyter_client            8.6.2              pyhd8ed1ab_0    conda-forge
jupyter_core              5.7.2            py39hcbf5309_0    conda-forge
jupyterlab_pygments       0.3.0              pyhd8ed1ab_1    conda-forge
keyring                   25.2.1             pyh7428d3b_0    conda-forge
kiwisolver                1.4.5            py39h1f6ef14_1    conda-forge
krb5                      1.21.2               heb0366b_0    conda-forge
lcms2                     2.16                 h67d730c_0    conda-forge
lerc                      4.0.0                h63175ca_0    conda-forge
libasprintf               0.22.5               h5728263_2    conda-forge
libasprintf-devel         0.22.5               h5728263_2    conda-forge
libblas                   3.9.0              22_win64_mkl    conda-forge
libbrotlicommon           1.1.0                hcfcfb64_1    conda-forge
libbrotlidec              1.1.0                hcfcfb64_1    conda-forge
libbrotlienc              1.1.0                hcfcfb64_1    conda-forge
libcblas                  3.9.0              22_win64_mkl    conda-forge
libclang13                18.1.7          default_h97ce8ae_0    conda-forge
libdeflate                1.20                 hcfcfb64_0    conda-forge
libffi                    3.4.2                h8ffe710_5    conda-forge
libgettextpo              0.22.5               h5728263_2    conda-forge
libgettextpo-devel        0.22.5               h5728263_2    conda-forge
libglib                   2.80.2               h7025463_1    conda-forge
libhwloc                  2.10.0          default_h8125262_1001    conda-forge
libiconv                  1.17                 hcfcfb64_2    conda-forge
libintl                   0.22.5               h5728263_2    conda-forge
libintl-devel             0.22.5               h5728263_2    conda-forge
libjpeg-turbo             3.0.0                hcfcfb64_1    conda-forge
liblapack                 3.9.0              22_win64_mkl    conda-forge
libogg                    1.3.4                h8ffe710_1    conda-forge
libpng                    1.6.43               h19919ed_0    conda-forge
libsodium                 1.0.18               h8d14728_1    conda-forge
libspatialindex           2.0.0                h5a68840_0    conda-forge
libsqlite                 3.46.0               h2466b09_0    conda-forge
libtiff                   4.6.0                hddb2be6_3    conda-forge
libvorbis                 1.3.7                h0e60522_0    conda-forge
libwebp                   1.4.0                h2466b09_0    conda-forge
libwebp-base              1.4.0                hcfcfb64_0    conda-forge
libxcb                    1.15                 hcd874cb_0    conda-forge
libxml2                   2.12.7               h283a6d9_1    conda-forge
libzlib                   1.3.1                h2466b09_1    conda-forge
m2w64-gcc-libgfortran     5.3.0                         6    conda-forge
m2w64-gcc-libs            5.3.0                         7    conda-forge
m2w64-gcc-libs-core       5.3.0                         7    conda-forge
m2w64-gmp                 6.1.0                         2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    conda-forge
markdown-it-py            3.0.0              pyhd8ed1ab_0    conda-forge
markupsafe                2.1.5            py39ha55989b_0    conda-forge
matplotlib                3.8.4            py39hcbf5309_2    conda-forge
matplotlib-base           3.8.4            py39he1095e7_2    conda-forge
matplotlib-inline         0.1.7              pyhd8ed1ab_0    conda-forge
mccabe                    0.7.0              pyhd8ed1ab_0    conda-forge
mdurl                     0.1.2              pyhd8ed1ab_0    conda-forge
mistune                   3.0.2              pyhd8ed1ab_0    conda-forge
mkl                       2024.1.0           h66d3029_692    conda-forge
more-itertools            10.3.0             pyhd8ed1ab_0    conda-forge
mpmath                    1.2.1              pyhd8ed1ab_0    conda-forge
msys2-conda-epoch         20160418                      1    conda-forge
multidict                 6.0.5            py39ha55989b_0    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
mypy_extensions           1.0.0              pyha770c72_0    conda-forge
nbclient                  0.10.0             pyhd8ed1ab_0    conda-forge
nbconvert                 7.16.4               hd8ed1ab_1    conda-forge
nbconvert-core            7.16.4             pyhd8ed1ab_1    conda-forge
nbconvert-pandoc          7.16.4               hd8ed1ab_1    conda-forge
nbformat                  5.10.4             pyhd8ed1ab_0    conda-forge
nest-asyncio              1.6.0              pyhd8ed1ab_0    conda-forge
numpy                     1.23.4           py39hbccbffa_1    conda-forge
numpydoc                  1.7.0              pyhd8ed1ab_1    conda-forge
openjpeg                  2.5.2                h3d672ee_0    conda-forge
openssl                   3.3.1                h2466b09_0    conda-forge
packaging                 24.1               pyhd8ed1ab_0    conda-forge
pandas                    1.5.1            py39h2ba5b7c_1    conda-forge
pandoc                    3.2                  h57928b3_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
paramiko                  3.4.0              pyhd8ed1ab_0    conda-forge
parso                     0.8.4              pyhd8ed1ab_0    conda-forge
pathspec                  0.12.1             pyhd8ed1ab_0    conda-forge
pcre2                     10.44                h3d7b363_0    conda-forge
pexpect                   4.9.0              pyhd8ed1ab_0    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    10.3.0           py39h9ee4981_0    conda-forge
pip                       24.0               pyhd8ed1ab_0    conda-forge
pkgutil-resolve-name      1.3.10             pyhd8ed1ab_1    conda-forge
platformdirs              4.2.2              pyhd8ed1ab_0    conda-forge
pluggy                    1.5.0              pyhd8ed1ab_0    conda-forge
ply                       3.11               pyhd8ed1ab_2    conda-forge
prompt-toolkit            3.0.47             pyha770c72_0    conda-forge
psutil                    5.9.8            py39ha55989b_0    conda-forge
pthread-stubs             0.4               hcd874cb_1001    conda-forge
pthreads-win32            2.9.1                hfa6e2cd_3    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
pycodestyle               2.11.1             pyhd8ed1ab_0    conda-forge
pyconify                  0.1.6              pyhd8ed1ab_0    conda-forge
pycparser                 2.22               pyhd8ed1ab_0    conda-forge
pydocstyle                6.3.0              pyhd8ed1ab_0    conda-forge
pyflakes                  3.2.0              pyhd8ed1ab_0    conda-forge
pygithub                  2.3.0              pyhd8ed1ab_0    conda-forge
pygments                  2.18.0             pyhd8ed1ab_0    conda-forge
pyjwt                     2.8.0              pyhd8ed1ab_1    conda-forge
pylint                    3.2.3              pyhd8ed1ab_0    conda-forge
pylint-venv               3.0.3              pyhd8ed1ab_0    conda-forge
pyls-spyder               0.4.0              pyhd8ed1ab_0    conda-forge
pynacl                    1.5.0            py39h09fa780_3    conda-forge
pyopenssl                 24.0.0             pyhd8ed1ab_0    conda-forge
pyparsing                 3.1.2              pyhd8ed1ab_0    conda-forge
pyqt                      5.15.9           py39hb77abff_5    conda-forge
pyqt5-sip                 12.12.2          py39h99910a6_5    conda-forge
pyqt6                     6.7.0                    pypi_0    pypi
pyqt6-qt6                 6.7.2                    pypi_0    pypi
pyqt6-sip                 13.6.0                   pypi_0    pypi
pyqtwebengine             5.15.9           py39h2f4a3f1_5    conda-forge
pysocks                   1.7.1              pyh0701188_6    conda-forge
pytest                    7.2.0              pyhd8ed1ab_2    conda-forge
pytest-cov                4.0.0              pyhd8ed1ab_0    conda-forge
pytest-lazy-fixture       0.6.3                      py_0    conda-forge
pytest-mock               3.10.0             pyhd8ed1ab_0    conda-forge
pytest-order              1.0.1              pyhd8ed1ab_0    conda-forge
pytest-qt                 4.2.0              pyhd8ed1ab_0    conda-forge
pytest-timeout            2.1.0              pyhd8ed1ab_0    conda-forge
python                    3.9.19          h4de0772_0_cpython    conda-forge
python-dateutil           2.9.0              pyhd8ed1ab_0    conda-forge
python-fastjsonschema     2.20.0             pyhd8ed1ab_0    conda-forge
python-gssapi             1.8.3            py39h0943f9b_0    conda-forge
python-lsp-black          2.0.0              pyhd8ed1ab_0    conda-forge
python-lsp-jsonrpc        1.1.2              pyhd8ed1ab_0    conda-forge
python-lsp-server         1.11.0             pyhd8ed1ab_0    conda-forge
python-lsp-server-base    1.11.0             pyhd8ed1ab_0    conda-forge
python-slugify            8.0.4              pyhd8ed1ab_0    conda-forge
python_abi                3.9                      4_cp39    conda-forge
pytoolconfig              1.2.5              pyhd8ed1ab_0    conda-forge
pytz                      2024.1             pyhd8ed1ab_0    conda-forge
pyuca                     1.2                        py_1    conda-forge
pywin32                   306              py39h99910a6_2    conda-forge
pywin32-ctypes            0.2.2            py39hcbf5309_1    conda-forge
pyyaml                    6.0.1            py39ha55989b_1    conda-forge
pyzmq                     26.0.3           py39h03e5c00_0    conda-forge
qdarkstyle                3.2.3              pyhd8ed1ab_0    conda-forge
qstylizer                 0.2.3              pyhd8ed1ab_0    conda-forge
qt-main                   5.15.8              h9e85ed6_20    conda-forge
qt-webengine              5.15.8               h4bf5c4e_4    conda-forge
qtawesome                 1.3.1              pyh9208f05_0    conda-forge
qtconsole                 5.6.0.dev0                dev_0    <develop>
qtpy                      2.4.1              pyhd8ed1ab_0    conda-forge
referencing               0.35.1             pyhd8ed1ab_0    conda-forge
requests                  2.32.3             pyhd8ed1ab_0    conda-forge
rich                      13.7.1             pyhd8ed1ab_0    conda-forge
rope                      1.13.0             pyhd8ed1ab_0    conda-forge
rpds-py                   0.18.1           py39h92a245a_0    conda-forge
rtree                     1.2.0            py39h7ad2cf7_1    conda-forge
scipy                     1.9.3            py39hfbf2dce_2    conda-forge
setuptools                70.0.0             pyhd8ed1ab_0    conda-forge
sip                       6.7.12           py39h99910a6_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
snowballstemmer           2.2.0              pyhd8ed1ab_0    conda-forge
sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
soupsieve                 2.5                pyhd8ed1ab_1    conda-forge
sphinx                    7.3.7              pyhd8ed1ab_0    conda-forge
sphinxcontrib-applehelp   1.0.8              pyhd8ed1ab_0    conda-forge
sphinxcontrib-devhelp     1.0.6              pyhd8ed1ab_0    conda-forge
sphinxcontrib-htmlhelp    2.0.5              pyhd8ed1ab_0    conda-forge
sphinxcontrib-jsmath      1.0.1              pyhd8ed1ab_0    conda-forge
sphinxcontrib-qthelp      1.0.7              pyhd8ed1ab_0    conda-forge
sphinxcontrib-serializinghtml 1.1.10             pyhd8ed1ab_0    conda-forge
spyder                    6.0.0b3.dev0             pypi_0    pypi
spyder-kernels            3.0.0.dev0                dev_0    <develop>
stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
superqt                   0.6.7              pyh9208f05_0    conda-forge
sympy                     1.11.1           py39hcbf5309_2    conda-forge
tabulate                  0.9.0              pyhd8ed1ab_1    conda-forge
tbb                       2021.12.0            hc790b64_1    conda-forge
text-unidecode            1.3                pyhd8ed1ab_1    conda-forge
textdistance              4.6.2              pyhd8ed1ab_0    conda-forge
three-merge               0.1.1              pyh9f0ad1d_0    conda-forge
tinycss2                  1.3.0              pyhd8ed1ab_0    conda-forge
tk                        8.6.13               h5226925_1    conda-forge
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
tomlkit                   0.12.5             pyha770c72_0    conda-forge
tornado                   6.4.1            py39ha55e580_0    conda-forge
traitlets                 5.14.3             pyhd8ed1ab_0    conda-forge
types-python-dateutil     2.9.0.20240316     pyhd8ed1ab_0    conda-forge
typing-extensions         4.12.2               hd8ed1ab_0    conda-forge
typing_extensions         4.12.2             pyha770c72_0    conda-forge
tzdata                    2024a                h0c530f3_0    conda-forge
ucrt                      10.0.22621.0         h57928b3_0    conda-forge
ujson                     5.10.0           py39ha51f57c_0    conda-forge
unicodedata2              15.1.0           py39ha55989b_0    conda-forge
urllib3                   2.2.2              pyhd8ed1ab_0    conda-forge
vc                        14.3                h8a93ad2_20    conda-forge
vc14_runtime              14.40.33810         ha82c5b3_20    conda-forge
vs2015_runtime            14.40.33810         h3bf8584_20    conda-forge
watchdog                  4.0.1            py39hcbf5309_0    conda-forge
wcwidth                   0.2.13             pyhd8ed1ab_0    conda-forge
webencodings              0.5.1              pyhd8ed1ab_2    conda-forge
whatthepatch              1.0.5              pyhd8ed1ab_0    conda-forge
wheel                     0.43.0             pyhd8ed1ab_1    conda-forge
win_inet_pton             1.1.0              pyhd8ed1ab_6    conda-forge
wrapt                     1.16.0           py39ha55989b_0    conda-forge
xorg-libxau               1.0.11               hcd874cb_0    conda-forge
xorg-libxdmcp             1.1.3                hcd874cb_0    conda-forge
xz                        5.2.6                h8d14728_0    conda-forge
yaml                      0.2.5                h8ffe710_2    conda-forge
yapf                      0.40.1             pyhd8ed1ab_0    conda-forge
yarl                      1.9.4            py39ha55989b_0    conda-forge
zeromq                    4.3.5                he1f189c_4    conda-forge
zipp                      3.19.2             pyhd8ed1ab_0    conda-forge
zstd                      1.5.6                h0ea2cb4_0    conda-forge
```

</details>

ccordoba12 commented:
Ok, if the problem is visible in master, it means it's not related to this PR. So, please open a new issue about it and tag it for rc1.
